src/Fmt_ast.ml

```diff
@@ -1502,15 +1501,19 @@ and fmt_expression c ?(box= true) ?epi ?eol ?parens ?ext
             $ fmt_expression c (sub_exp ~ctx e2) )
         $ fmt_atrs )
   | Pexp_tuple es ->
-      let parens =
+      let force_no_parens =
         match xexp.ctx with
-        | Str {pstr_desc= Pstr_eval _} -> false
-        | _ -> true
```

Comment:
I'm nervous about "manually" forcing no parens here. I think that the effect of not emitting discretionary/redundant parens around tuple expressions can be achieved by changing this `true` to `parens`. Probably a few little things also need to be sorted out (like `assoc_of_prec` should be `Non` on `Comma` I expect), but doing it this way would use the precedence-based parenthesization algorithm uniformly with the rest of the code. So far the approach has been to (try to) systematically ensure that the parenthesization functions in Ast return safe results, which are sometimes overridden in Fmt_ast to add _more_ parens in ad hoc ways.